File: MaskedEditText/src/main/java/com/vicmikhailau/maskededittext/MaskedFormatter.java
package com.vicmikhailau.maskededittext;


public class MaskedFormatter {
    Mask mMask;

    MaskedFormatter() {
        mMask = null;
    }

    public MaskedFormatter(String fmtString) {
        this();
        this.setMask(fmtString);
    }


    public String getMaskString() {
        if (mMask != null) {
            return mMask.getFormatString();
        }
        return null;
    }

    public int getMaskLength() {
        return mMask.size();
    }

    public void setMask(String fmtString) {
        mMask = new Mask(fmtString);
    }


    public IFormattedString formatString(String value) {
        return mMask.getFormattedString(value);
    }

}

File: MaskedEditText/src/main/java/com/vicmikhailau/maskededittext/MaskedFormatter.kt
package com.vicmikhailau.maskededittext


class MaskedFormatter internal constructor() {
    internal var mMask: Mask? = null


    val maskString: String?
        get() = mMask?.formatString

    val maskLength: Int?
        get() = mMask?.size()

    init {
        mMask = null
    }

    constructor(fmtString: String) : this() {
        this.setMask(fmtString)
    }

    fun setMask(fmtString: String) {
        mMask = Mask(fmtString)
    }


    fun formatString(value: String): IFormattedString? {
        return mMask?.getFormattedString(value)
    }

}
